RestaurantReservation.API/Controllers/AuthinticateController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
+using RestaurantReservation.API.Models.Authinticate;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+
+namespace RestaurantReservation.API.Controllers
+{
+    [ApiController]
+    [Route("api/auth")]
+    public class AuthinticateController : Controller
+    {
+        private readonly CustomerRepository _customerRepository;
```

Comment:
For all repositories, I see that you are not using the DI principle (Dependency Inversion).
It's really bad and should be avoided whenever possible.
Try writing unit tests for this controller and see how much pain it would be. 

**Note: this comment applies to all repositories and their usages.**